querybook/server/logic/board_permission.py

```diff
@@ -11,21 +12,26 @@ class BoardDoesNotExist(Exception):
 
 @with_session
 def user_can_edit(board_id, uid, session=None):
-    board = Board.get(id=board_id, session=session)
+    board, editor = session.query(Board, BoardEditor).outerjoin(
+        BoardEditor,
+        and_(Board.id == BoardEditor.board_id, BoardEditor.uid == uid),
+    ).filter(Board.id == board_id).first() or (None, None)
 
     if board is None:
         raise BoardDoesNotExist()
 
     if board.owner_uid == uid:
         return True
 
-    # TODO: Update with edit permissions
-    return False
+    return editor is not None and editor.write
 
 
 @with_session
 def user_can_read(board_id, uid, session=None):
-    board = Board.get(id=board_id, session=session)
+    board, editor = session.query(Board, BoardEditor).outerjoin(
+        BoardEditor,
+        and_(Board.id == BoardEditor.board_id, BoardEditor.uid == uid),
+    ).filter(Board.id == board_id).first() or (None, None)
```

Comment:
Don't know if join is necessary here either. Can also create a function to get the board editor by uid & board_id since it's used here for a second time.